feat: init charter (#1)
webpack/governance: CHARTER.md

- Optimization of assets for production, including minification, compression, and tree shaking (removal of unused code)
- Enabling hot module replacement (HMR) for faster development feedback loops
- Configuration and extensibility through plugins and loaders
- Support for code splitting and lazy loading to enhance performance
- Integrating with modern JavaScript frameworks (e.g., React, Vue, Angular)
- Providing detailed build reports and debugging tools for development and production
- Supporting multiple output formats (e.g., AMD, CommonJS, ES Modules) for compatibility with various environments
- Community-driven tools and best practices provided through
  [plugins](https://webpack.js.org/plugins/) and [guides](https://webpack.js.org/guides/)
- Technical help and discussions via community platforms such as
  [GitHub Discussions](https://github.com/webpack/webpack/discussions)

In addition to the main repository, `webpack` (bundler), the organization maintains and manages several other core
projects that are integral to the ecosystem such as `webpack-dev-server`, `webpack-cli`,  and `webpack-contrib` projects.

> These repositories, along with others housed within the `github.com/webpack` and `github.com/webpack-contrib are owned by the webpack project.`.

Reviewer: I'm not sure if this is on purpose
```suggestion
These repositories, along with others housed within the `github.com/webpack` and `github.com/webpack-contrib` are owned by the webpack project.
```

Author: When we wrote this, we wanted examples of which projects that is ours, but I think this is better. LGTM

resignation or by consensus with at least five voting members to be present. A vote in a TSC meeting can be used to change a regular TSC member to a voting
TSC member, or to change a voting TSC member to a regular TSC member.

No more than one-fourth of the TSC voting members may be affiliated with the same employer. If a change in TSC voting
membership or a change of employment by a TSC voting member creates a situation where more than one-fourth of the TSC
voting membership shares an employer, then the situation must be immediately remedied by the removal of voting member
status from one or more TSC voting members affiliated with the over-represented employer(s).

The TSC shall meet regularly using tools that enable participation by the community (e.g. weekly on a Google Hangout On
Air, or through any other appropriate means selected by the TSC). The meeting shall be directed by the TSC Chairperson.
Responsibility for directing individual meetings may be delegated by the TSC Chairperson to any other TSC voting member.
Minutes or an appropriate recording shall be taken and made available to the community through accessible public
postings.

> TSC voting members are expected to regularly participate in TSC meetings and issue triaging. Meeting notes will be released such that absent members of the TSC can provide feedback in any way.

Reviewer 2: ```suggestion
TSC voting members are expected to regularly participate in TSC meetings and issue triaging.
```

Seems like the second part of that paragraph is a little redundant with the minute taking noted above.

within the webpack project, including:

* Setting release dates.
* Release quality standards.
* Technical direction.
* Project governance and process.
* GitHub repository management.
* Conduct and maintain guidelines defined by the OpenJS Foundation Cross Project Council.
* Maintaining the list of additional collaborators.
* Development process and any coding standards.
* Mediating technical conflicts between collaborators or `webpack` projects.

The TSC voting members will define webpack project's release vehicles.

### Section 4.1. webpack Project Operations

The TSC voting members will establish and maintain a development process for the webpack project. The development
process will establish guidelines for how the developers and community will operate. It will, for example, establish
appropriate timelines for TSC review (e.g. agenda items must be published at least a certain number of hours in advance
of a TSC meeting).

Note that project operations remain subject to any relevant policy or process specified by the OpenJS Foundation board or Cross Project Council.

### Section 4.2. Decision-making, Voting, and/or Elections

#### Section 4.2.1. Elections

> These processes are described in webpack's [GOVERNANCE](GOVERNANCE.md) document.

Reviewer 2: It seems that the election process is actually described in section 3 above, no?

Maybe move it here?

Reviewer 3: I don't see anything related to elections on section 3 🤔

Reviewer 2: Well, this is where you describe how TSC members are chosen, so that’s the closest you have to elections.

Reviewer 3: Let me see if I understood this correctly: you're asking for us to move this election sub-section to be part of section 3?

(The charter template on the CPC's repo mentions elections as part of section 4, that's why I added here) -- because there's no reference in section 3 of how someone becomes a TSC member.

#### Section 4.2.2. Decision Making

For webpack projects' decisions, Collaborators and TSC voting members shall operate under Lazy Consensus ([consensus seeking][]). When an agenda item has appeared to reach a
consensus, the moderator will ask "Does anyone object?" as a final call for dissent from the consensus. For all votes, a simple majority of all TSC voting members for, or against, the issue wins. A TSC voting member may
choose to participate in any vote through abstention. Votes are needed to add new members, remove members from the TSC or deciding in project critical issues, such as roadmapping major versions of webpack core.

## Section 5. Definitions

> * **Contributors**: contribute code or other artifacts, but do not have the right to commit to the code base.

Reviewer 2: The term isn't used anywhere in the charter. No point in defining it at all.

Reviewer 3: I guess we can remove it in the opposite direction. Is there a term that would make sense to be added to the Definitions, in your opinion?

Reviewer 2: Not that I think of.

- Supporting multiple output formats (e.g., AMD, CommonJS, ES Modules) for compatibility with various environments
- Community-driven tools and best practices provided through 
  [plugins](https://webpack.js.org/plugins/) and [guides](https://webpack.js.org/guides/)
- Technical help and discussions via community platforms such as 
  [GitHub Discussions](https://github.com/webpack/webpack/discussions)

In addition to the main repository, `webpack` (bundler), the organization maintains and manages several other core 
projects that are integral to the ecosystem such as:

- **webpack-dev-server**: Provides a local server with live reloading and hot module replacement, improving developer 
  productivity.
- **webpack-cli**: Offers a command-line interface to interact with webpack's build processes and configurations.
- **webpack-contrib projects**: Includes various loaders, plugins, middlewares and utility repositories such as:
  - `sass-loader`
  - `file-loader`
  - `style-loader`
  - `webpack-hot-middleware`
  - `webpack-bundle-analyzer`

These enable the handling of different asset types, interacting with webpack in various capacities to monitor builds 
and improve applications.

> These projects, along with others housed within the `github.com/webpack` and `github.com/webpack-contrib`.

Reviewer 3: @evenstensberg I think we forgot the Out-of-scope section? Was it removed?

- Supporting multiple output formats (e.g., AMD, CommonJS, ES Modules) for compatibility with various environments
- Community-driven tools and best practices provided through 
  [plugins](https://webpack.js.org/plugins/) and [guides](https://webpack.js.org/guides/)
- Technical help and discussions via community platforms such as 
  [GitHub Discussions](https://github.com/webpack/webpack/discussions)

In addition to the main repository, `webpack` (bundler), the organization maintains and manages several other core 
projects that are integral to the ecosystem such as:

- **webpack-dev-server**: Provides a local server with live reloading and hot module replacement, improving developer 
  productivity.
- **webpack-cli**: Offers a command-line interface to interact with webpack's build processes and configurations.
- **webpack-contrib projects**: Includes various loaders, plugins, middlewares and utility repositories such as:
  - `sass-loader`
  - `file-loader`
  - `style-loader`
  - `webpack-hot-middleware`
  - `webpack-bundle-analyzer`

These enable the handling of different asset types, interacting with webpack in various capacities to monitor builds 

> and improve applications.

Reviewer 2: If the list isn't comprehensive, maybe shorten it, e.g.:
```suggestion
In addition to the main repository, `webpack` (bundler), the organization maintains and manages several other core 
projects that are integral to the ecosystem such as `webpack-dev-server`, `webpack-cli`,  and `webpack-contrib` projects.
```

Reviewer 3: That's a good call, simplifying the list makes sense.

Author: fixed.

> These projects, along with others housed within the `github.com/webpack` and `github.com/webpack-contrib`.

Reviewer 2: This sentence is unclear.

Reviewer 3: ```suggestion
These repositories, along with others housed within the `github.com/webpack` and `github.com/webpack-contrib are owned by the webpack project.`.
```

Reviewer 3: Something like that?

Author: fixed

### 1.2: Out-of-Scope

- Testing frameworks (e.g., unit, integration, or end-to-end testing tools)

> - API development and handling

Reviewer 2: Can you be a little more specific about API dev? is is web APIs?

Reviewer 3: @evenstensberg you wrote this entry, what did you have in mind with this? 🤔

Author: I meant external tools like CLIs tapping into webpack

Author: fixed

- Server-side rendering directly (although it can integrate with SSR frameworks)

> - Frameworks and libraries that operate in their own capacity.

Reviewer 2: ```suggestion
- `webpack`-related frameworks and libraries that operate in their own capacity.
```

Author: fixed.

## Section 2. Relationship with OpenJS Foundation CPC.

Most large, complex open source communities have both a business and a technical governance model. Technical leadership 
for the projects within the OpenJS Foundation is delegated to the projects through their project charters by the OpenJS 
Cross Project Council (CPC). In the case of the webpack project, it is delegated to the webpack Technical Steering 
Committee ("TSC"). OpenJS Foundation's business leadership is the Board of Directors (the "Board").

This charter can only be amended with the approval of the CPC.

## Section 3. Establishment of the TSC

TSC members can be either _regular_ members or _voting_ members. Regular members can attend meetings and participate in 
TSC discussions, but do not vote. Voting members can do everything regular members can do, and also have the ability to 
cast votes when consensus is not reached on an issue.

TSC memberships are not time-limited. There is no maximum size of the TSC. The TSC must have at least four voting 
members.

> There is no specific set of requirements or qualifications for TSC membership beyond these rules. The TSC may add 

Reviewer 2: what does "these rules" mean here? Can you rephrase that sentence to make it more clear?

Author: we removed it, not critical.

additional voting members to the TSC by a standard TSC motion. A TSC member can be removed from the TSC by voluntary 

> resignation or by a standard TSC motion. A standard TSC motion can be used to change a regular TSC member to a voting 

Reviewer 2: What is "a standard TSC motion"? Where is it defined?

Author: fixed by adding more specific terms.

TSC member, or to change a voting TSC member to a regular TSC member.

No more than one-fourth of the TSC voting members may be affiliated with the same employer. If a change in TSC voting 
membership or a change of employment by a TSC voting member creates a situation where more than one-fourth of the TSC 
voting membership shares an employer, then the situation must be immediately remedied by the removal of voting member 
status from one or more TSC voting members affiliated with the over-represented employer(s).

The TSC shall meet regularly using tools that enable participation by the community (e.g. weekly on a Google Hangout On 
Air, or through any other appropriate means selected by the TSC). The meeting shall be directed by the TSC Chairperson. 
Responsibility for directing individual meetings may be delegated by the TSC Chairperson to any other TSC voting member. 
Minutes or an appropriate recording shall be taken and made available to the community through accessible public 
postings.

> TSC voting members are expected to regularly participate in TSC activities.

Reviewer 2: What are TSC activities? Maybe define them more precisely.

Author: fixed.

A TSC voting member is automatically converted to a TSC regular member if they do not participate in three consecutive 
TSC votes.

## Section 4. Roles & Responsibilities of the TSC

Subject to such policies as may be set by the CPC, the TSC voting members are responsible for all technical development 
within the webpack project, including:

* Setting release dates.
* Release quality standards.
* Technical direction.
* Project governance and process.

> * GitHub repository hosting.

Reviewer 2: I'm not sure what "GitHub repository hosting" means here

Author: fixed.

> * Conduct guidelines.

Reviewer 2: This isn't super clear. The CPC is responsible for the code of conduct. The TSC might have opted in to handle CoC reports. If it's the latter you'll want to clarify this.

Reviewer 3: I believe this bullet list was a copy of the current Node.js's ones.

Author: fixed.

> * Maintaining the list of additional Collaborators.

Reviewer 2: Collaborators aren't defined as a term. Either define it or lowercase it. (note you can lowercase even if you define it, that makes the text easier to read and more approachable.)

Author: fixed.

* Development process and any coding standards.

> * Mediating technical conflicts between Collaborators or Foundation projects.

Reviewer 2: ```suggestion
* Mediating technical conflicts between Collaborators or `webpack` projects.
```

Author: fixed.

The TSC voting members will define webpack project's release vehicles.

### Section 4.1. webpack Project Operations

The TSC voting members will establish and maintain a development process for the webpack project. The development 
process will establish guidelines for how the developers and community will operate. It will, for example, establish 
appropriate timelines for TSC review (e.g. agenda items must be published at least a certain number of hours in advance 
of a TSC meeting).

> The TSC and entire technical community will follow any processes as may be specified by the OpenJS Foundation Board 

Reviewer 2: I'd frame this as more of a note. e.g.:

"Note that project operations remain subject to any relevant policy or process specified by the board or the CPC."

Author: fixed.

relating to the intake and license compliance review of contributions, including the OpenJS Foundation IP Policy.

### Section 4.2. Decision-making, Voting, and/or Elections

#### Section 4.2.1. Elections

Leadership roles in the webpack project will be peer elected representatives of the community.

For election of persons (such as the TSC Chairperson), a multiple-candidate method should be used, such as:

* [Condorcet][] or
* [Single Transferable Vote][]

Multiple-candidate methods may be reduced to simple election by plurality when there are only two candidates for one 
position to be filled. No election is required if there is only one candidate and no objections to the candidate's 

> election. Elections shall be done within the projects by the Collaborators active in the project.

Reviewer 2: It looks like you're mixing up seats and candidates here. I suggest simplifying this section and copying it from what the CPC does for elections.

Author: @ovflowd can you find the relevant text we need to change to?

Reviewer 3: Sure!

> election. Elections shall be done within the projects by the Collaborators active in the project.

Reviewer 2: "Elections shall be done within the projects by the Collaborators active in the project." Totally unclear why would webpack projects elect people and what their roles would be. I suggest entirely removing this last sentence and moving a webpack project level governance outside of the charter.

Reviewer 3: Will leave them intentionally blank and let this be decided afterwards as part of the project governance outside of the charter.

> The TSC voting members will elect from amongst voting TSC members a TSC Chairperson to work on building an agenda for 

Reviewer 2: Don't you need two reps?

TSC meetings and a OpenJS Cross Project Council (CPC) voting member to represent the TSC in the OpenJS Foundation for a 
term of one year. The Chair and voting CPC member may be (but are not required to be) the same person. The TSC shall 
hold annual elections to select a TSC Chairperson and voting CPC member; there are no limits on the number of terms a 
TSC Chairperson or voting CPC member may serve.

> #### Section 4.2.2. Voting

Reviewer 2: Call this section decision-making

Author: fixed.

> For internal project decisions, Collaborators shall operate under Lazy Consensus. The TSC voting members shall establish 

Reviewer 2: Define "internal projects" somewhere or better yet, consistently call them "webpack projects."

Author: fixed.

appropriate guidelines for implementing Lazy Consensus (e.g. expected notification and review time periods) within the 
development process.

> The TSC voting members follow a [Consensus Seeking][] decision making model. When an agenda item has appeared to reach a 

Reviewer 2: is "lazy consensus" and "Consensus Seeking decision making model" the same or different?

If it's the same, use the same words. If it is different define both and explain how they differ.

Author: fixed.

consensus the moderator will ask "Does anyone object?" as a final call for dissent from the consensus.

> For all votes, a simple majority of all TSC voting members for, or against, the issue wins. A TSC voting member may 

Reviewer 2: When do you vote on things? you want to spell this out.

Author: fixed.

choose to participate in any vote through abstention.

> All changes to this charter must be approved by the CPC.

Reviewer 2: What is this doing in the section on voting?

Author: fixed.

> #### Section 4.2.3. Project Roles

Reviewer 2: ```suggestion
#### Section 4.3. Other Project Roles
```

Author: We removed this

> The webpack git repository is maintained by the TSC and additional Collaborators who are added by the TSC voting members 

Reviewer 2: What about other subprojects? This feels like an implementation detail that doesn't belong in the charter

Author: fixed by removing the section.

on an ongoing basis.

Individuals making significant and valuable contributions are made Collaborators and given commit-access to the project.
These individuals are identified by the TSC and their addition as Collaborators is discussed during a TSC meeting. 
Modifications of the contents of the git repository are made on a collaborative basis as defined in the development 
process.

> Collaborators may opt to elevate significant or controversial modifications, or modifications that have not found 
> consensus to the TSC for discussion by assigning the `tsc-agenda` tag to a pull request or issue. The TSC voting members 
> should serve as the final arbiter where required. The TSC will maintain and publish a list of current Collaborators, as 
> well as a development process guide for Collaborators and Contributors looking to participate in the development effort.

Reviewer 2: Don't put implementation details in the charter
```suggestion
Collaborators may opt to elevate significant or controversial modifications, or modifications that have not found 
consensus to the TSC for discussion. The TSC voting members 
should serve as the final arbiter where required. The TSC will maintain and publish a list of current Collaborators, as 
well as a development process guide for Collaborators and Contributors looking to participate in the development effort.
```

Author: We removed this section

> well as a development process guide for Collaborators and Contributors looking to participate in the development effort.

Reviewer 2: "development process guide" / "development process"; if it's the same doc, use consistent terminology.

Author: We use `development process` throughout the doc.

## Section 5. Definitions

* **Contributors**: contribute code or other artifacts, but do not have the right to commit to the code base. 
  Contributors work with the project's Collaborators to have code committed to the code base. A Contributor may be 

>   promoted to a Collaborator by the TSC. Contributors should rarely be encumbered by the TSC and never by the CPC or 

Reviewer 2: encumbered feels like an unfortunate term, here. tbh.

Author: fixed

  OpenJS Foundation Board.

> * **Project**: a technical collaboration effort, e.g. a subsystem, that is organized through the project creation 

Reviewer 2: Use `webpack project` or `internal project` here to distinguish it from what the foundation calls projects.

Author: fixed.

>   process and approved by the TSC.

Reviewer 2: what is the "project creation process" ?

Author: fixed.
